src/pages/DataAccessRequestApplication.js

```diff
@@ -184,12 +186,22 @@ class DataAccessRequestApplication extends Component {
     return datasets;
   }
 
+  async getSOs()  {
+    const allSOs = await User.getSOsForCurrentUser();
+    const allSOsOptions = map((user) =>  {
+      return {value: user.displayName, label: user.displayName};
+    })(allSOs);
+    return allSOsOptions;
+  }
+
   async init() {
     const { dataRequestId } = this.props.match.params;
     let formData = {};
     const researcher = await User.getMe();
+    const signingOfficials = await this.getSOs();
     this.setState(prev => {
       prev.researcher = researcher;
+      prev.allSigningOfficials = signingOfficials;
```

Comment:
I think this is an acceptable approach. I think the problem we had in the earlier iteration was that the api call was inside a `useEffect` that was bundled with a lot of other dependencies. If instead it were pulled out into its own separate `useEffect` with an empty array of dependencies, then there would be no triggers on re-render, and it would not re-call the api every time a change happened.

Reply:
Re-requesting review now that the options load and save as expected